backend/src/db/repositories/PostRepository.ts

```diff
@@ -83,28 +83,46 @@ export default class PostRepository {
             });
     }
 
-    async getPostsByUser(userId: number, forUserId: number, page: number, perPage: number): Promise<PostRawWithUserData[]> {
+    async getPostsByUser(userId: number, forUserId: number, filter: string, page: number, perPage: number): Promise<PostRawWithUserData[]> {
         const limitFrom = (page - 1) * perPage;
-
-        return await this.db.query(`
+        let query = `
                 select p.*, v.vote, b.read_comments, b.bookmark, b.last_read_comment_id, b.watch
                 from posts p
                          left join post_votes v on (v.post_id = p.post_id and v.voter_id = :for_user_id)
                          left join user_bookmarks b on (b.post_id = p.post_id and b.user_id = :for_user_id)
                 where p.author_id = :user_id
                 order by created_at desc
                     limit :limit_from, :limit_count
-            `,
-            {
-                user_id: userId,
-                for_user_id: forUserId,
-                limit_from: limitFrom,
-                limit_count: perPage
-            });
+            `;
+        const params = {
+            user_id: userId,
+            for_user_id: forUserId,
+            limit_from: limitFrom,
+            limit_count: perPage
+        };
+        if (filter) {
+            query = `select p.*, v.vote, b.read_comments, b.bookmark, b.last_read_comment_id, b.watch
```

Comment:
Same as above, let's try to avoid the query duplication.

Reply:
I am unsure whether we should use it here and I am debating with myself, but I tend to lean towards not using it. This would duplicate advanced search functionality once it's there